src/sync/sync_session.cpp

```diff
@@ -298,16 +298,18 @@ void SyncSession::create_sync_session()
         }
 
         using ProtocolError = realm::sync::ProtocolError;
+        SyncError error = SyncError{error_code, std::move(message), is_fatal};
 
-        SyncSessionError error_type = SyncSessionError::Debug;
         // Precondition: error_code is a valid realm::sync::Error raw value.
-        ProtocolError strong_code = static_cast<ProtocolError>(error_code);
+        ProtocolError strong_code = static_cast<ProtocolError>(error_code.value());
```

Comment:
This cast is not safe. `std::error_code` may contain errors that are not of the category belonging to `ProtocolError`.